src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,232 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+
+/**
+ * The Tasklist class that stores the arraylist.
+ */
+public class TaskList {
+
+	private static Pattern CHECKSTRING = Pattern.compile("-?\\d+");
+
+	private ArrayList<Task> list;
+
+	/**
+	 * Constructor of a tasklist
+	 */
+	public TaskList() {
+		this.list = new ArrayList<Task>();
+	}
+
+	/**
+	 * To check if an integer is given after user input "mark", "unmark" or "delete".
+	 *
+	 * @param strNum The String after user input "mark", "unmark" or "delete".
+	 * @return True if it is an integer, false otherwise.
+	 */
+	private static boolean isInteger(String strNum) {
+		if (strNum == null) {
+			return false;
+		}
+		return CHECKSTRING.matcher(strNum).matches();
+	}
+
+	/**
+	 * Adds task from the text document to the arraylist.
+	 *
+	 * @param task The task from the text document.
+	 */
+
+	public void addStorageToList(Task task) {
+		this.list.add(task);
+	}
+
+	/**
+	 * gets index from the arraylist.
+	 *
+	 * @param number The inndex
+	 * @return The task in that index.
+	 */
+	public Task getTask(int number) {
+		return this.list.get(number);
+	}
+
+	/**
+	 * Gets the length of the arraylist.
+	 *
+	 * @return The length of the arraylist
+	 */
+	public int size() {
+		return this.list.size();
+	}
+
+	/**
+	 * Clears the arraylist.
+	 */
+	public void clear() {
+		this.list.clear();
+	}
+
+	/**
+	 * marks task with an X
+	 *
+	 * @param task
+	 */
+	public static void markString(Task task) {
+		task.mark();
+		System.out.println("-----------------------------------------------");
+		System.out.println("Nice! I've marked this task as done:");
+		System.out.println(task);
+		System.out.println("-----------------------------------------------");
+	}
+
+	/**
+	 * Unmarks task by removing the X.
+	 *
+	 * @param task
+	 */
+	public static void unMarkString(Task task) {
+		task.unMark();
+		System.out.println("-----------------------------------------------");
+		System.out.println("OK, I've marked this task as not done yet:");
+		System.out.println(task);
+		System.out.println("-----------------------------------------------");
```

Comment:
Perhaps you can consider migrating these print statements into Ui.java as well for a better organisation of classes.